data-analytics/postgres-to-bigquery/src/main/python/postgres_to_bigquery_utils.py

```diff
@@ -0,0 +1,346 @@
+                              "WHERE table_schema = '{0}' and table_name ='{1}' " \
+                              "ORDER BY ordinal_position) table_json) TO STDOUT".format(schema_name, table_name)
+        get_json_data_sql_raw = "COPY (SELECT row_to_json(data_result) FROM (SELECT {} FROM {}.{}) data_result) TO STDOUT"
+        get_all_columns_sql = "SELECT column_name,data_type FROM information_schema.columns " \
+                              "WHERE table_schema = '{0}' AND table_name='{1}' ORDER BY ordinal_position" \
+            .format(schema_name, table_name)
+
+        connection = connection.get_conn()
+        cur = connection.cursor()
+        cur.execute(get_all_columns_sql)
+        columns_list = cur.fetchall()
+        columns_list_string = ""
+        for column in columns_list:
+            if column[1] != 'timestamp with time zone':
+                columns_list_string += ('"' + column[0] + '",')
+            else:
+                columns_list_string += ("EXTRACT(EPOCH FROM " + column[0] + ") AS " + column[0] + ",")
+
+        # Remove last comma
+        columns_list_string = columns_list_string[:-1]
+        get_json_data_sql = get_json_data_sql_raw.format(columns_list_string, schema_name, table_name)
+
+        schema_file_name = local_schema_dir + table_name + "_schema.json"
+        data_file_name = local_data_dir + table_name + "_data.json"
+
+        # Write Schema file
+        f = open(schema_file_name, 'wb')
+        f.write(bytes("[\n", "UTF-8"))
+        cur.copy_expert(get_json_schema_sql, f)
+        f.seek(-2, 2)  # remove last comma
+        f.write(bytes("]\n", "UTF-8"))
+        f.close()
+        logging.info("Write JSON schema to file " + schema_file_name)
+        # Remove extra \ character
+        ret = subprocess.call(['sed', '-i', '-e', r"s/\\\\/\\/g", schema_file_name])
+        if (ret == 0):
+            logging.info("Writing schema file " + schema_file_name + " successfully!")
+        else:
+            logging.error("Having trouble removing extra characters in file: " + schema_file_name)
+
+        # Write data file
+        df = open(data_file_name, "wb")
+        cur.copy_expert(get_json_data_sql, df)
+        df.close()
+
+        connection.close()
+
+        ret = subprocess.call(['sed', '-i', '-e', r"s/\\\\/\\/g", data_file_name])
+        if (ret == 0):
+            logging.info("Writing data file " + data_file_name + " successfully!")
+        else:
+            logging.error("Having trouble removing extra characters in file: " + data_file_name)
+
+        # Upload to GCS and remove from local
+        subprocess.call(['gsutil', '-m', 'cp', schema_file_name, gcs_schema_uri])
+        logging.info("Write to GCS bucket: " + schema_file_name + " successfully!")
+        subprocess.call(
+            ['gsutil', '-o GSUtil:parallel_composite_upload_threshold=150M', '-m', 'cp', data_file_name, gcs_data_uri])
+        logging.info("Write to GCS bucket: " + data_file_name + " successfully!")
```

Comment:
You never actually check the return value from the call to gsutil. Check the value, provide an error message if something went wrong, remove exclamation mark from your log messages, use %s with arg list in your logging calls.